p4_symbolic/bmv2/test.cc

```diff
@@ -43,34 +36,36 @@ void WriteFile(char path[], const std::string& content) {
 }
 
 // The main test routine for parsing bmv2 json with protobuf.
-// Parses bmv2 json that is fed in through stdin and dumps
-// the resulting native protobuf and json data to files.
-// Expects the paths of the protobuf output file and json
-// output file to be passed as command line arguments respectively.
+// Parses bmv2 json file and dumps the resulting bmv2 protobuf
+// and json data to files.
+// Expects the paths of the input json file and the protobuf and json output
+// files to be passed as command line arguments in order.
 int main(int argc, char* argv[]) {
   // Verify link and compile versions are the same.
   GOOGLE_PROTOBUF_VERIFY_VERSION;
 
   // Validate command line arguments.
-  if (argc != 3) {
-    std::cout << "Usage: ./main <protobuf output file> <json output file>."
+  if (argc != 4) {
+    std::cout << "Usage: ./main <input JSON file> <protobuf output file> <json "
+                 "output file>."
               << std::endl;
     return 0;
   }
 
-  // Read input json from stdin.
-  std::string input = ReadStdin();
-
-  // Parsing JSON with protobuf.
-  p4_symbolic::bmv2::P4Program p4_buf;
-  google::protobuf::util::JsonParseOptions parsing_options;
-  parsing_options.ignore_unknown_fields = true;
-  google::protobuf::util::JsonStringToMessage(input, &p4_buf, parsing_options);
+  // Parse JSON using bmv2.cc.
+  std::string input(argv[1]);
+  pdpi::StatusOr<p4_symbolic::bmv2::P4Program> status =
+      p4_symbolic::bmv2::ParseBmv2Json(input);
```

Comment:
The convention is to use absl::string_view for string arguments. If you do this for `ParseBmv2Json` here, you don't need to convert char* to std::string.

Reply:
https://abseil.io/tips/1

Reply:
I am going to keep this in mind but will put it on the back burner for a bit

Reply:
Sounds good. I opened an issue.

Reply:
I am going to open an issue for this, and leave this for the next step. I would like to merge this PR to keep things simple as I add new things.

Reply:
Oops, you are ahead of me!